src/nl/hannahsten/texifyidea/refactoring/inlinefile/LatexInlineFileProcessor.kt

```diff
@@ -0,0 +1,86 @@
+package nl.hannahsten.texifyidea.refactoring.inlinefile
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.command.CommandProcessor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiReference
+import com.intellij.psi.search.GlobalSearchScope
+import com.intellij.psi.search.SearchScope
+import com.intellij.psi.search.searches.ReferencesSearch
+import com.intellij.refactoring.BaseRefactoringProcessor
+import com.intellij.usageView.UsageInfo
+import com.intellij.usageView.UsageViewDescriptor
+import nl.hannahsten.texifyidea.file.LatexFile
+
+/**
+ * Recieves settings and a target and performs the requested refactoring
+ *
+ * @see com.intellij.refactoring.inline.InlineMethodProcessor
+ *
+ * @author jojo2357
+ */
+class LatexInlineFileProcessor(
+    myProject: Project,
+    private val inlineFile: LatexFile,
+    private val originalReference: PsiElement?,
+    private val isInlineThisOnly: Boolean,
+    private val isKeepTheDeclaration: Boolean,
+    private val myScope: SearchScope = GlobalSearchScope.projectScope(myProject)
+) : BaseRefactoringProcessor(myProject) {
+
+    override fun createUsageViewDescriptor(usages: Array<out UsageInfo>): UsageViewDescriptor {
+        return LatexInlineFileDescriptor(inlineFile)
+    }
+
+    override fun findUsages(): Array<UsageInfo> {
+        val tempreferences = ReferencesSearch.search(inlineFile, myScope).findAll().asSequence()
+
+        return tempreferences
+            .distinct()
+            .map(PsiReference::getElement)
+            .map(::UsageInfo)
+            .toList()
+            .toTypedArray()
+    }
+
+    override fun performRefactoring(usages: Array<out UsageInfo>) {
+        if (isInlineThisOnly) {
+            if (originalReference != null)
+                replaceUsage(originalReference)
+            // else error case
```

Comment:
Error case?

Reply:
This should never happen since to get here you have to select inline this only from the dialog that should never have let you select that outside of the editor. It is such an edge case that I am not sure if an exception is needed or just a log

Reply:
Added an exception if it ever happens outside a unit test